Plugins/InstallSafeDITool/InstallCLIPluginCommand.swift

```diff
@@ -111,3 +103,72 @@ struct InstallSafeDITool: CommandPlugin {
         }
     }
 }
+
+#if canImport(XcodeProjectPlugin)
+    import XcodeProjectPlugin
+
+    extension InstallSafeDITool: XcodeCommandPlugin {
+        func performCommand(
+            context: XcodeProjectPlugin.XcodePluginContext,
+            arguments _: [String]
+        ) throws {
+            let version = context.safeDIVersion
+            let safediFolder = context.safediFolder
+            let expectedToolFolder = context.expectedToolFolder
+            let expectedToolLocation = context.expectedToolLocation
+
+            #if arch(arm64)
+                let toolName = "SafeDITool-arm64"
+            #elseif arch(x86_64)
+                let toolName = "SafeDITool-x86_64"
+            #else
+                Diagnostics.error("Unexpected architecture type")
+                exit(1)
+            #endif
+
+            let githubDownloadURL = context.safeDIOrigin.appending(
+                components: "releases",
+                "download",
+                version,
+                toolName
+            )
+
+            let dispatchGroup = DispatchGroup()
+            dispatchGroup.enter()
+            Task.detached {
+                defer { dispatchGroup.leave() }
+                let (downloadedURL, _) = try await URLSession.shared.download(
+                    for: URLRequest(url: githubDownloadURL)
+                )
+                let downloadedFileAttributes = try FileManager.default.attributesOfItem(atPath: downloadedURL.path())
+                guard let currentPermissions = downloadedFileAttributes[.posixPermissions] as? NSNumber,
+                      // Add executable attributes to the downloaded file.
+                      chmod(downloadedURL.path(), mode_t(currentPermissions.uint32Value) | S_IXUSR | S_IXGRP | S_IXOTH) == 0
+                else {
+                    Diagnostics.error("Failed to make downloaded file \(downloadedURL.path()) executable")
+                    exit(1)
+                }
+                try FileManager.default.createDirectory(
+                    at: expectedToolFolder,
+                    withIntermediateDirectories: true
+                )
+                try FileManager.default.moveItem(
+                    at: downloadedURL,
+                    to: expectedToolLocation
+                )
+                let gitIgnoreLocation = safediFolder.appending(component: ".gitignore")
+                if !FileManager.default.fileExists(atPath: gitIgnoreLocation.path()) {
+                    try """
+                    */\(expectedToolLocation.lastPathComponent)
+                    """.write(
+                        to: gitIgnoreLocation,
+                        atomically: true,
+                        encoding: .utf8
+                    )
+                }
+            }
+            // Force the command to wait until the async work is done.
+            dispatchGroup.wait()
```

Comment:
this is nuts, but since the method is not `async` (unlike the plugin for Package.swift files), we have to pull this trick to ensure that the `main` method here only exits after all our work is done.